src/components/webcam-texture-target.js

```diff
@@ -0,0 +1,84 @@
+import { disposeTexture } from "../utils/material-utils";
+
+AFRAME.registerComponent("webcam-texture-target", {
+  schema: {
+    src: { type: "string" }
+  },
+
+  getMaterial() {
+    return (
+      (this.el.object3DMap.skinnedmesh && this.el.object3DMap.skinnedmesh.material) ||
+      (this.el.object3DMap.mesh && this.el.object3DMap.mesh.material) ||
+      this.el.object3D.material
+    );
+  },
+
+  init() {
+    const material = this.getMaterial();
+    this.originalTexture = material && material.map;
+  },
+
+  update(prevData) {
+    const material = this.getMaterial();
+
+    console.log({ src: this.data.src, material, prevSrc: prevData.src });
+
+    if (!material) {
+      return;
+    }
+
+    const src = this.data.src;
+
+    if (prevData.src === src) {
+      return;
+    }
+
+    if (src && src.startsWith("hubs://")) {
+      const streamClientId = src.substring(7).split("/")[1]; // /clients/<client id>/video is only URL for now
+
+      this.originalTexture = material.map;
+
+      NAF.connection.adapter.getMediaStream(streamClientId, "video").then(stream => {
+        if (src !== this.data.src) {
+          // Prevent creating and loading video texture if the src changed while we were fetching the video stream.
+          return;
+        }
+
+        const video = document.createElement("video");
+        video.setAttribute("playsinline", "");
+        video.setAttribute("webkit-playsinline", "");
+        // iOS Safari requires the autoplay attribute, or it won't play the video at all.
+        video.autoplay = true;
+        // iOS Safari will not play videos without user interaction. We mute the video so that it can autoplay and then
+        // allow the user to unmute it with an interaction in the unmute-video-button component.
+        video.muted = AFRAME.utils.device.isIOS();
+        video.preload = "auto";
+        video.crossOrigin = "anonymous";
+        video.srcObject = new MediaStream(stream.getVideoTracks());
```

Comment:
The stream already should just be video tracks so I think you can skip wrapping it in a new stream. If it did have audio tracks you arguably might want to play those anyway.